src/NHibernate/Transform/AliasToBeanResultTransformer.cs

```diff
@@ -121,6 +121,10 @@ public override IList TransformList(IList collection)
 		/// same visibility and inheritance depth.</exception>
 		private void SetProperty(string alias, object value, object resultObj)
 		{
+			if (alias == null)
+				// Grouping properties in criteria are selected without alias, just ignore them.
+				return;
```

Comment:
The fix. Reintroduces the check removed [here](https://github.com/nhibernate/nhibernate-core/pull/597/commits/9491f57aca2264e293889dc960e49ddfb20350f8#diff-e431ffcf2f1f91c671a967d74bcb1252L85) without us spotting it and questioning it.